pkg/plugins/manager/loader/loader.go

```diff
@@ -54,6 +61,15 @@ func (l *Loader) Load(ctx context.Context, src plugins.PluginSource) ([]*plugins
 		return nil, err
 	}
 
+	names := make([]string, len(initializedPlugins))
+	for i, p := range initializedPlugins {
+		names[i] = p.ID
+	}
+	sort.Strings(names)
+	plugins := strings.Join(names, ", ")
+
+	sourceLogger.Debug("Plugin source loaded", "plugins", plugins, "duration", time.Since(start))
```

Comment:
probably a better place for this is pkg/plugins/manager/pipeline/initialization/initialization.go since it's referring to initialization (and we are already looping over plugins there)

Also, since you are already calculating the time in the `Store` service, I don't think you need to do it here too?

Reply:
Initialize is happening at the last stage of the pipeline and I wanted better instrumentation around the whole pipeline. The time allows us to understand if loading times increase/decrease per source why I thought it was a good idea to have them in multiple places. WDYT?

Reply:
ah, okay so you want:

 - Total time, regardless of the source (already in Store).
 - Time per source. This can also be moved to Store, within the loop.
 - List of plugins loaded. This can be moved to Initialize.

My main reasoning is to avoid adding things to `Load` which are out of a step in the pipeline. It just a nitpick but it helps keeping this function neat.

Reply:
> Time per source. This can also be moved to Store, within the loop.

I did have it in the store initially, but since you can install plugin during runtime I reckoned adding it to the loader would allow both the store and installer to log when loading a source. If I move it back need to add it to installer as well. Can do that if you think it makes more sense?

> List of plugins loaded. This can be moved to Initialize.

Tricky to get/pass the start time/duration by doing this.

Would it be more suitable if I wrap the loader then in a loaderLogger struct that implements Load and calls wrapped loader.Load?

Reply:
okay, I get your point. Up to you if you want to leave it here. 

> > List of plugins loaded. This can be moved to Initialize.
> 
> Tricky to get/pass the start time/duration by doing this.

I was talking about printing just the list of plugins loaded, not the time. What I am not completely happy of this is looping over plugins, sorting, etc. People may continue adding logic to this function outside a pipeline step, which is what we don't want to encourage.

Another suggestion would be doing something like this at the beginning:

```go
	end := instrumentLoad()

	discoveredPlugins, err := l.discovery.Discover(ctx, src)
	...
	initializedPlugins, err := l.initializer.Initialize(ctx, validatedPlugins)
	if err != nil {
		return nil, err
	}
	
	end(initializedPlugins)
```

At least, the scope is more neat.

Reply:
Good idea, I'll change